src/gif.rs

```diff
@@ -387,7 +387,13 @@ impl<W: Write> Encoder<W> {
 
         // Create the gif::Frame from the animation::Frame
         let mut frame = Frame::from_rgba(width, height, &mut *rbga_frame);
-        frame.delay = (frame_delay / 10).try_into().map_err(|_|ImageError::DimensionError)?;
```

Comment:
Here the `Option::None` is being mapped into a DimensionError if the `frame_delay / 10` expression of type `u32` doesn't fit into a `u16`. I believe this should have a different error kind returned as it is not really dimension related?

Reply:
That's precisely why we're doing these, to uncover the misuse in some places.

Reply:
There are two ways to go about this, I'm not quite sure which one is better.

* Treat this as a saturating conversion and use `u16::MAX`. This doesn't sound too bad given that the current algorithm is lossy in any case and then this would justifiably be the closest possible approximation. This is only relevant for frame durations above 10 minutes in any case.
* .. OR add a new variant to the `ParameterErrorKind` here.

Reply:
I would be in favor of the saturating conversion, as you already said that the algorithm is already lossy and because a new `ParameterErrorKind` for this seems like it would be solely used for this specific instance. At least I can't think of a name that could possibly have more applications than this instance.

Reply:
Agreed then, be sure to add a comment with the explanation for why the saturation is performed.